cgi/product_image_upload.pl

```diff
@@ -314,6 +315,8 @@
 			and (  (not defined single_param('source'))
 				or (single_param('source') ne "product_edit_form")
 				or (not defined $product_ref->{images}{$imagefield}))
+			and (not is_protected_image($product_ref, $imagefield) or $User{moderator})
```

Comment:
We need to add a similar check on 2 other files:

product_image_crop.pl : it is called when you edit a product page on the web, and try to crop / select another image.

product_image_unselect.pl : used to unselect a selected image